streams/src/main/java/org/apache/kafka/streams/kstream/PrintForeachAction.java

```diff
@@ -34,13 +34,30 @@
      * @param streamName The given name will be printed.
      */
     public PrintForeachAction(final PrintWriter printWriter, final String streamName) {
+        this(printWriter, null, streamName);
+    }
+
+    /**
+     * Print customized output with given writer. The PrintWriter can be null in order to
+     * distinguish between {@code System.out} and the others. If the PrintWriter is {@code PrintWriter(System.out)},
+     * then it would close {@code System.out} output stream.
+     * <p>
+     * Afterall, not to pass in {@code PrintWriter(System.out)} but {@code null} instead.
+     *
+     * @param printWriter Use {@code System.out.println} if {@code null}.
+     * @param mapper The mapper which can allow user to customize output will be printed.
+     * @param streamName The given name will be printed.
+     */
+    public PrintForeachAction(final PrintWriter printWriter, KeyValueMapper<? super K, ? super V, String> mapper, final String streamName) {
         this.printWriter = printWriter;
```

Comment:
I already have null check for KeyValueMapper in `PrintForeachAction.apply` and will use default format if null.